invenio_drafts_resources/services/records/service.py

```diff
@@ -375,6 +376,15 @@ def publish(self, identity, id_, uow=None, expand=False):
         """
         # Get the draft
         draft = self.draft_cls.pid.resolve(id_, registered_only=False)
+
+        # If config is True and there are no communities selected
+        # Then, check for permissions to upload without community
+        if (
+            current_app.config["RDM_RECORD_ALWAYS_IN_COMMUNITY"]
+            and len(draft.parent.communities.ids) == 0
+        ):
+            if not self.check_permission(identity, "publish", record=draft):
```

Comment:
to improve the readability:
```suggestion

        community_required = current_app.config["RDM_RECORD_ALWAYS_IN_COMMUNITY"]
        community_missing = len(draft.parent.communities.ids) == 0
        can_publish = self.check_permission(identity, "publish", record=draft)
        
        if (
            community_required and community_missing and not can_publish
        ):
```

also not ideal, in some cases we check the permissions for publish twice (second time in line 388)